stdlib/SuiteSparse/src/cholmod.jl

```diff
@@ -804,7 +804,15 @@ function Dense{T}(A::StridedVecOrMat) where T<:VTypes
     end
     d
 end
-function Dense(A::StridedVecOrMat)
+function Dense{T}(A::Union{Adjoint{<:Any, <:StridedVecOrMat}, Transpose{<:Any, <:StridedVecOrMat}}) where T<:VTypes
+    d = allocate_dense(size(A, 1), size(A, 2), size(A, 1), T)
+    s = unsafe_load(pointer(d))
+    for i in 1:length(A)
+        unsafe_store!(s.x, A[i], i)
```

Comment:
I have modified this code, so that it uses `eachindex` now. As far as I understand this, it should not depend on being one-indexed anymore.

Reply:
I was actually thinking that it does, via `unsafe_store!`, and that the same applies to the method above that you haven’t touched. In that sense, I thought your approach was „more correct“ than what we have above. But I’m not familiar with `unsafe_store!` and `unsafe_load`.

Reply:
In the initial version
```julia
for i in 1:length(A)
    unsafe_store!(s.x, A[i], i)
end
```
I have assumed that the array `A` is 1-indexed, because I have accessed `A` (inside the loop) at `1`, `2`, ..., `length(A)`.

In the current version
```julia
for (i, c) in enumerate(eachindex(A))
    unsafe_store!(s.x, A[c], i)
end
```
I access `A` at indices given by the `eachindex` function, so they should be valid, even if `A` is not 1-indexed.

Reply:
I understand all that. But assume that the first index is `-100`. Then I’m not sure that `unsafe_store!` will work correctly, at least judging by its docstring.

Reply:
It seems to work fine for such cases:
```julia
julia> using OffsetArrays

julia> import SuiteSparse.CHOLMOD: allocate_dense

julia> A = OffsetArray{Float64}(undef, -100:-95, -100:-95)
       for (i,c) in enumerate(eachindex(A))
            A[c] = i
       end

julia> A
6×6 OffsetArray(::Array{Float64,2}, -100:-95, -100:-95) with eltype Float64 with indices -100:-95×-100:-95:
 1.0   7.0  13.0  19.0  25.0  31.0
 2.0   8.0  14.0  20.0  26.0  32.0
 3.0   9.0  15.0  21.0  27.0  33.0
 4.0  10.0  16.0  22.0  28.0  34.0
 5.0  11.0  17.0  23.0  29.0  35.0
 6.0  12.0  18.0  24.0  30.0  36.0

julia> B = A'
6×6 LinearAlgebra.Adjoint{Float64,OffsetArray{Float64,2,Array{Float64,2}}} with indices -100:-95×-100:-95:
  1.0   2.0   3.0   4.0   5.0   6.0
  7.0   8.0   9.0  10.0  11.0  12.0
 13.0  14.0  15.0  16.0  17.0  18.0
 19.0  20.0  21.0  22.0  23.0  24.0
 25.0  26.0  27.0  28.0  29.0  30.0
 31.0  32.0  33.0  34.0  35.0  36.0

julia> d = allocate_dense(6, 6, 6, Float64)
       GC.@preserve d begin
           s = unsafe_load(pointer(d))
           for (i,c) in enumerate(eachindex(B))
               unsafe_store!(s.x, B[c], i)
           end
       end

julia> d
6×6 SuiteSparse.CHOLMOD.Dense{Float64}:
  1.0   2.0   3.0   4.0   5.0   6.0
  7.0   8.0   9.0  10.0  11.0  12.0
 13.0  14.0  15.0  16.0  17.0  18.0
 19.0  20.0  21.0  22.0  23.0  24.0
 25.0  26.0  27.0  28.0  29.0  30.0
 31.0  32.0  33.0  34.0  35.0  36.0
```
@dkarrasch Do you think further changes are needed here?